numba/tests/test_svml.py

```diff
@@ -345,96 +328,81 @@ def check(self, pyfunc, *args, **kwargs):
         np.testing.assert_almost_equal(jitstd_result, py_expected, **kwargs)
         np.testing.assert_almost_equal(jitfast_result, py_expected, **kwargs)
 
+    def check_asm(self, pyfunc, *args, **kwargs):
+        std_pattern = kwargs.pop('std_pattern', None)
+        fast_pattern = kwargs.pop('fast_pattern', None)
+
         # look for specific patterns in the asm for a given target
-        with override_env_config('NUMBA_CPU_NAME', cpu_name), \
-             override_env_config('NUMBA_CPU_FEATURES', cpu_features):
-            # recompile for overridden CPU
-            jitstd, jitfast = self.compile(pyfunc, *args)
-            if std_pattern:
-                self.check_svml_presence(jitstd, std_pattern)
-            if fast_pattern:
-                self.check_svml_presence(jitfast, fast_pattern)
+        # recompile for overridden CPU
+        jitstd, jitfast = self.compile(pyfunc, *args)
+        if std_pattern:
+            self.check_svml_presence(jitstd, std_pattern)
+        if fast_pattern:
+            self.check_svml_presence(jitfast, fast_pattern)
+
+    def check(self, pyfunc, *args, what="both", **kwargs):
+        assert what in ("both", "result", "asm")
+        if what == "both" or what == "result":
+            self.check_result(pyfunc, *args, **kwargs)
+        if what == "both" or what == "asm":
+            self.check_asm(pyfunc, *args, **kwargs)
 
     def check_svml_presence(self, func, pattern):
         asm = func.library.get_asm_str()
         self.assertIn(pattern, asm)
 
-    def test_scalar_context(self):
+    @TestCase.run_test_in_subprocess(envvars=_skylake_axv512_envvars)
+    def test_scalar_context_asm(self):
         # SVML will not be used.
         pat = '$_sin' if config.IS_OSX else '$sin'
-        self.check(math_sin_scalar, 7., std_pattern=pat)
-        self.check(math_sin_scalar, 7., fast_pattern=pat)
+        self.check(math_sin_scalar, 7., what="asm", std_pattern=pat)
+        self.check(math_sin_scalar, 7., what="asm", fast_pattern=pat)
 
-    def test_svml(self):
+    def test_scalar_context_result(self):
+        # checks result for test_scalar_context_asm
+        self.check(math_sin_scalar, 7., what="result")
+        self.check(math_sin_scalar, 7., what="result")
```

Comment:
why run twice?

Reply:
Copy paste error I guess, fixed in 5bd7254.